app/controllers/charges_controller.rb

```diff
@@ -0,0 +1,26 @@
+class ChargesController < ApplicationController
+  def new
+  end
+
+  def create
+    @pending_purchase = PendingPurchase.find(params[:pending_purchase_id])
```

Comment:
It looks like this instance variable isn't being used in any views, so you should make it a local variable instead.